account_move_line_purchase_info/migrations/14.0.1.0.0/pre-migration.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    """
+    Column `purchase_id` of table `account_move_line` has been renamed to `purchase_order_id`
+    because `purchase_order_id` is now in the core.
+    """
+    old_purchase_column = "purchase_id"
+    new_purchase_column = "purchase_order_id"
+    move_line_table_name = "account_move_line"
+    ou_purchase_rename = openupgrade.get_legacy_name(old_purchase_column)
+
+    if openupgrade.column_exists(env.cr, move_line_table_name, ou_purchase_rename):
+        openupgrade.rename_columns(
+            env.cr,
+            {
+                move_line_table_name: [(ou_purchase_rename, new_purchase_column)],
+            },
+        )
```

Comment:
```suggestion

    if not openupgrade.column_exists(env.cr, move_line_table_name, new_purchase_column):
        openupgrade.rename_fields(env, [
    ('account.move.line', 'account_move_line', old_purchase_column, new_purchase_column),
])
```

Reply:
@MiquelRForgeFlow  Thank you for your suggestion. I already update this file.